spec/cc/engine/analyzers/ruby/main_spec.rb

```diff
@@ -43,10 +45,43 @@ module CC::Engine::Analyzers
         expect(json["other_locations"]).to eq([
           {"path" => "foo.rb", "lines" => { "begin" => 9, "end" => 13} },
         ])
+
         expect(json["content"]["body"]).to match /This issue has a mass of `36`/
         expect(json["fingerprint"]).to eq("f21b75bbd135ec3ae6638364d5c73762")
       end
 
+      it "creates an issue for each occurrence of the duplicated code" do
+        create_source_file("foo.rb", <<-EORUBY)
+            describe '#ruby?' do
+              before { subject.type = 'ruby' }
+
+              it 'returns true' do
+                expect(subject.ruby?).to be true
+              end
+            end
+
+            describe '#js?' do
+              before { subject.type = 'js' }
+
+              it 'returns true' do
+                expect(subject.js?).to be true
+              end
+            end
+
+            describe '#whaddup?' do
+              before { subject.type = 'js' }
+
+              it 'returns true' do
+                expect(subject.js?).to be true
+              end
+            end
+        EORUBY
+
+        issues = run_engine(engine_conf).strip.split("\0")
+
+        expect(issues.length).to eq(3)
```

Comment:
I think we need to get some better coverage, and can probably do so with a spec on `Violations`. Here you assert we get three `Violation`s out, which is good -- but I'd like to also confirm some attributes on each violation, that the main and other locations are accurate mostly.